oracle/oracle.go

```diff
@@ -444,3 +458,125 @@ func (o *oracleOps) Delete(volumeID string) error {
 	}
 	return nil
 }
+
+func (o *oracleOps) SetInstanceGroupSize(instanceGroupID string, count int64, timeout time.Duration) error {
+
+	ctx := context.Background()
+	var cancel context.CancelFunc
+	if timeout > time.Nanosecond {
+		// If timeout is non-zero then
+		// add deadline to the operation else
+		// wait infinitely for operation to complete
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
+
+	instanceGroupSize := int(count)
+
+	//get containerengine client
+	client, err := containerengine.NewContainerEngineClientWithConfigurationProvider(common.DefaultConfigProvider())
+	if err != nil {
+		return err
+	}
+
+	//get nodepool by ID to be updated
+	nodePoolReq := containerengine.ListNodePoolsRequest{CompartmentId: &o.compartmentID, Name: &instanceGroupID, ClusterId: &o.clusterID}
+	nodePools, err := client.ListNodePools(context.Background(), nodePoolReq)
+	if err != nil {
+		return err
+	}
+
+	if len(nodePools.Items) == 0 {
+		return errors.New("No node pool found with name " + instanceGroupID)
+	}
+	numberOfDomains := len(nodePools.Items[0].NodeConfigDetails.PlacementConfigs)
+	totalClusterSize := numberOfDomains * instanceGroupSize
+	logrus.Println("Setting instanceGroupSize to ", totalClusterSize, " in total ", numberOfDomains, " regions.")
+
+	//get all availabliity domain
+	nodePoolPlacementConfigDetails := make([]containerengine.NodePoolPlacementConfigDetails, numberOfDomains)
+
+	for i, placementConfigs := range nodePools.Items[0].NodeConfigDetails.PlacementConfigs {
+		nodePoolPlacementConfigDetails[i].AvailabilityDomain = placementConfigs.AvailabilityDomain
+		nodePoolPlacementConfigDetails[i].SubnetId = placementConfigs.SubnetId
+	}
+
+	//update node pools
+	req := containerengine.UpdateNodePoolRequest{
+		NodePoolId: nodePools.Items[0].Id, //get node pool id
+		UpdateNodePoolDetails: containerengine.UpdateNodePoolDetails{
+			NodeConfigDetails: &containerengine.UpdateNodePoolNodeConfigDetails{
+				Size:             &totalClusterSize,
+				PlacementConfigs: nodePoolPlacementConfigDetails,
```

Comment:
Here, we are not making any changes to placementConfig and it's not a required (must) field. We can skip creating a duplicate `nodePoolPlacementConfigDetails`, if oracle SDK allows that.

Reply:
nodePoolPlacementConfigDetails is required to equally balance creation and deletion of instances in each availability domain. We need to provide one placement configuration for each availability domain in which we intend to launch a node.